examples/gno.land/p/demo/int256/arithmetic.gno

```diff
@@ -5,23 +5,26 @@ import "gno.land/p/demo/uint256"
 func (z *Int) Add(x, y *Int) *Int {
 	z.initiateAbs()
 
-	neg := x.neg
-
 	if x.neg == y.neg {
-		// x + y == x + y
-		// (-x) + (-y) == -(x + y)
-		z.abs = z.abs.Add(x.abs, y.abs)
+		// If both numbers have the same sign, add their absolute values
+		z.abs.Add(x.abs, y.abs)
+		z.neg = x.neg
 	} else {
-		// x + (-y) == x - y == -(y - x)
-		// (-x) + y == y - x == -(x - y)
+		// If signs are different, subtract the smaller absolute value from the larger
 		if x.abs.Cmp(y.abs) >= 0 {
-			z.abs = z.abs.Sub(x.abs, y.abs)
+			z.abs.Sub(x.abs, y.abs)
+			z.neg = x.neg
 		} else {
-			neg = !neg
-			z.abs = z.abs.Sub(y.abs, x.abs)
+			z.abs.Sub(y.abs, x.abs)
+			z.neg = y.neg
 		}
```

Comment:
suggestion:

```go
	switch x.abs.Cmp(y.abs) {
	case 1:
		z.abs.Sub(x.abs, y.abs)
		z.neg = x.neg
	case -1:
		z.abs.Sub(y.abs, x.abs)
		z.neg = y.neg
	case 0:
		z.abs.Clear()
		z.neg = false
	}
```

you can also remove the `if` below, then.

Reply:
https://github.com/gnolang/gno/pull/2750/commits/4e84f641f04c224a421aa5c0f85494b21ad2cb36